### PR TITLE
Feature/upgrade seastar api v6

### LINKED ISSUE #374: feature: upgrade seastar api to v6
**Upgrade seastar api to v6 version.**

Api migration will follow the line of: 
[https://github.com/vectorizedio/redpanda](https://github.com/vectorizedio/redpanda).
Progress tracked on: 
[https://github.com/lumontec/smf/tree/feature/upgrade_seastar_api_v6](https://github.com/lumontec/smf/tree/feature/upgrade_seastar_api_v6)

### PR DESCRIPTION
**Pull request by lumontec**
closes #374 

# Big changes
* **upgrade to seastar v6**: aligned cmake dependency to redpanda version git hash d80beb8f0e5f926b79db9538e62c32c5d3bcc076  
* **upgrade hwloc**: aligned to redpanda version v2.2
* **removed std-compat.h**
*  **refactor make_file_output_stream**: function now returns a future

# Minor changes
* **add valgrind to deps** required by seastar v6
* **other minor header fixes**

### CODE REVIEW

**Comment by senior7515** [comment]
This looks really good! 

can you sign the commits with git. 

we use DCO (https://elinux.org/Developer_Certificate_Of_Origin) 

so `git commit -s ...` or ammend + `-s`

File: CMakeLists.txt.in [hunk 1]
```diff
@@ -275,13 +275,17 @@ ExternalProject_Add(lksctp-tools
   INSTALL_COMMAND ${make_command} install)
 
 ExternalProject_Add(seastar
-  URL https://github.com/vectorizedio/seastar/archive/319d359.tar.gz
-  URL_MD5 341f15559319f93463961db03978d45c
+  GIT_REPOSITORY https://github.com/vectorizedio/seastar.git
+  GIT_TAG d80beb8f0e5f926b79db9538e62c32c5d3bcc076
```
Comment (senior7515) [inline CMakeLists.txt.in:279]: fantastic, i wonder if we copy all the other libs that `vectorizedio/redpanda` are using too.

File: CMakeLists.txt.in [hunk 2]
```diff
@@ -83,7 +83,7 @@ ExternalProject_Add(protobuf
 set(dpdk_quadruple ${CMAKE_SYSTEM_PROCESSOR}-native-linuxapp-gcc)
 
 set (dpdk_args
-  EXTRA_CFLAGS=-Wno-error
+  "EXTRA_CFLAGS=-Wno-error -fcommon"
```
Comment (senior7515) [inline CMakeLists.txt.in]: what does `-fcommon` do 

this seems OK, but typically a change in flags, would be it's own commit.

File: install-deps.sh [hunk 3]
```diff
@@ -66,6 +66,7 @@ function debs() {
         pkg-config \
         xfslibs-dev \
         systemtap-sdt-dev \
+	valgrind \
```
Comment (senior7515) [inline install-deps.sh:69]: is the indentation correct? missing 2 spaces i think

File: CMakeLists.txt.in [hunk 4]
```diff
@@ -294,7 +294,7 @@ ExternalProject_Add(seastar
     -DSeastar_TESTING=OFF
     -DSeastar_CXX_FLAGS=-Wno-error
     -DSeastar_LD_FLAGS=-pthread
-    -DSeastar_API_LEVEL=1
+    -DSeastar_API_LEVEL=2
```
Comment (senior7515) [inline CMakeLists.txt.in]: this is fantastic! i cannot believe smf still had the compat namespace. perfect. thank you.

**Review by senior7515 — COMMENTED** [review]
Overall this is fantastic! really good work. 

only minor nits left. can you sign the commits and I can then merge.

File: CMakeLists.txt.in (hunk 1 above)
Comment (lumontec) [reply CMakeLists.txt.in:279]: For sure ! I will handle that in a separate issue.

File: CMakeLists.txt.in (hunk 2 above)
Comment (lumontec) [reply CMakeLists.txt.in]: You`re right, that actually slipped inside the commit from redpanda, I cleaned that up with few other bits in [https://github.com/smfrpc/smf/pull/375/commits/713a5ae4c7519198d0c537ad1f0fec06695e6642](https://github.com/smfrpc/smf/pull/375/commits/713a5ae4c7519198d0c537ad1f0fec06695e6642)

It`s just a performance flag for the linker [https://man7.org/linux/man-pages/man1/gcc.1.html](https://man7.org/linux/man-pages/man1/gcc.1.html), not really required imho.

**Comment by lumontec** [comment]
Should be all good now !

**Review by senior7515 — APPROVED** [review]
LGTM.

**Comment by senior7515** [comment]
@lumontec fantastic! thank you so much.
